src/apiClient.ts

```diff
@@ -6,30 +6,37 @@ export interface ApiClientConfig {
   optionServiceUrl?: string
   analyticsServiceUrl?: string
   userNotificationServiceUrl?: string
-  tokenRefreshServiceUrl?: string
   debug?: boolean
 }
 
-export interface Tokens {
-  accessToken?: string
-  refreshToken?: string
-}
-
 export interface Handlers {
   onAccessTokenUpdate?: (token: string) => void
   onFailedTokenRefresh?: () => void
 }
 
 class ApiClient {
-  static instance: ApiClient
+  private static instance: ApiClient
 
-  configuration: ApiClientConfig = {}
-  tokens: Tokens = {}
-  handlers: Handlers = {}
-  version = "v1"
+  configuration: ApiClientConfig
+  accessToken: string | null
+  handlers: Handlers
+  version: string
+  refreshToken: () => Promise<{ accessToken: string }>
 
   constructor() {
+    if (ApiClient.instance) return ApiClient.instance
```

Comment:
could I have instantiated two api clients with different configurations before and can I still do this now? I'm not asking for that capability, but it might be worth thinking about whether we'd ever want this. or at least it's a breaking change worth pointing out

Reply:
I am not sure whether I understood the intention on the existing implementation properly but I assumed that having a singelton for the api client is intended and there was a flaw in the implementation
Maybe @Averethel can clarify on this?

Reply:
We still can have a singleton in a node process with a server-side configuration (think pre-fetching) and a client-side one, right?

Reply:
yes, that's what I see too